### PR TITLE
Use string/byte Replacers to avoid allocation/function calls.

### PR DESCRIPTION
**Pull request by nibbleshift**
Avoid allocating/calling byte/string replace multiple times by using Replacer objects.

### CODE REVIEW

File: internal/bloblang/query/methods_strings.go [hunk 1]
```diff
@@ -1465,22 +1470,19 @@ func replaceAllManyImpl(args *ParsedParams) (simpleMethod, error) {
...
-				t = strings.ReplaceAll(t, pair[0], pair[1])
-			}
-			return t, nil
+			return strReplacer.Replace(t), nil
 		case []byte:
-			for _, pair := range replacePairsBytes {
-				t = bytes.ReplaceAll(t, pair[0], pair[1])
-			}
-			return t, nil
+			return byteReplacer.Replace(t), nil
```
Comment (nibbleshift) [inline internal/bloblang/query/methods_strings.go:1485]: @Jeffail Should I clone `t` here?  The byte replace is make in-place modifications to `t`.

**Comment by peczenyj** [comment]
Hey @nibbleshift do you think you can solve the conflict?

**Comment by nibbleshift** [comment]
Hey @peczenyj, I'll clean it up this weekend.

**Comment by nibbleshift** [comment]
Fixed merge conflicts and removed out of scope dot path caching.

This branch only implements string replacers now.

**Comment by nibbleshift** [comment]
@Jeffail  Do you mind giving this a quick review?

**Pull request closed without merging**